client/request.go

```diff
@@ -104,12 +104,8 @@ func (cli *Client) buildRequest(method, path string, body io.Reader, headers htt
 	req = cli.addHeaders(req, headers)
 
 	if cli.proto == "unix" || cli.proto == "npipe" {
-		// For local communications, it doesn't matter what the host is. We just
-		// need a valid and meaningful host name. For details, see:
-		//
-		// - https://github.com/docker/engine-api/issues/189
-		// - https://github.com/golang/go/issues/13624
-		req.Host = "docker"
+		// For local communications, it doesn't matter what the host is.
+		req.Host = DummyHost
```

Comment:
Using an empty string for `req.Host` without any other changes has no effect because `req.URL.Host` will be used as the value of the Host request header.

Reply:
Ah, dang.

It was getting late here, and I thought to do a quick 'n dirty push to let CI run. If you have time feel free to push to my branches, or otherwise I'll continue Tomorrow